### PR TITLE
[FEATURE] Added new jsbase theme filled only with javascript and cha…

### PR DESCRIPTION
**Pull request by wolv-dev**
…nged extension of responsive theme, removed js from responsive theme

## Description
Added new Theme including only the javascript files. This enhances the process of creating new themes based on responsive theme, cause you don't need to copy and paste all the JS from one theme to another.
It increases maintainability of custom themes, cause the JS will be updated from you (Shopware) and no need to recopy and paste the updates JS in own custom theme.

I also updated the karma/jasmine tests to work with the new theme and added the dev dependencies for karma, jasmine, phantomJS, ... 

| Questions        | Answers
| ---------------- | -------------------------------------------------------
| BC breaks?       | no
| Tests pass?      | yes
| Related tickets? | No related ticket
| How to test?     | npm install in new BaseJs Theme, grunt and npm run test

### CODE REVIEW

**Comment by wolv-dev** [comment]
For questions reach out to me in IRC or Twitter with username "WolvDev"

File: themes/Frontend/JsBase/.gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Prevent committing npm modules or logs
+node_modules
+npm-debug.log
+# Prevent committing phantomJS JUnit Report
+TESTS*
```
Comment (xf-) [inline themes/Frontend/JsBase/.gitignore]: newline missing

**Comment by schliflo** [comment]
I cannot second that enough. Would make Shopware updates a lot easier for custom themes.

File: themes/Frontend/JsBase/.gitignore (hunk 1 above)
Comment (wolv-dev) [reply themes/Frontend/JsBase/.gitignore]: Existing .gitignore in Responsive Theme has no newline, too
https://github.com/shopware/shopware/blob/5.2/themes/Frontend/Responsive/.gitignore

Comment (schliflo) [reply themes/Frontend/JsBase/.gitignore]: perfect opportunity to add one i'd say ;)

File: themes/Frontend/JsBase/Gruntfile.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+                'Directory "<%= directory %>" was created successfully.',
+                { data: { directory: vendorDir } }
+            ));
+        }
+    });
+
+    grunt.registerTask('default', [ 'clean', 'createVendorDir', 'copy' ]);
+    grunt.loadNpmTasks('grunt-contrib-clean');
+    grunt.loadNpmTasks('grunt-contrib-copy');
+};
```
Comment (xf-) [inline themes/Frontend/JsBase/Gruntfile.js]: newline

File: themes/Frontend/JsBase/LICENSE-MIT [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
```
Comment (xf-) [inline themes/Frontend/JsBase/LICENSE-MIT]: newline and normally license files are without suffix. GitHub recognizes _none_, md and txt

File: themes/Frontend/JsBase/.gitignore (hunk 1 above)
Comment (wolv-dev) [reply themes/Frontend/JsBase/.gitignore]: Added

File: themes/Frontend/JsBase/Gruntfile.js (hunk 2 above)
Comment (wolv-dev) [reply themes/Frontend/JsBase/Gruntfile.js]: Added

File: themes/Frontend/JsBase/LICENSE-MIT (hunk 3 above)
Comment (wolv-dev) [reply themes/Frontend/JsBase/LICENSE-MIT]: Newline added.
Filename not changed, cause Shopware has done the same naming.

File: themes/Frontend/JsBase/package.json [hunk 4]
```diff
@@ -0,0 +1,47 @@
+{
+  "name": "shopwarebasjstheme",
```
Comment (shyim) [inline themes/Frontend/JsBase/package.json]: I guess you would write shopwarebasejstheme

File: themes/Frontend/JsBase/Theme.php [hunk 5]
```diff
@@ -0,0 +1,111 @@
+<?php
+
```
Comment (shyim) [inline themes/Frontend/JsBase/Theme.php]: Shopware Header Part is missing (https://github.com/shopware/shopware/blob/5.2/engine/Shopware/Application.php#L2)

File: themes/Frontend/JsBase/Theme.php [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    /**
+     * Allows to define a description text
+     * for the theme
+     * @var null
+     */
+    protected $description = '__theme_description__';
+
+    /**
+     * Name of the theme author.
+     * @var null
```
Comment (shyim) [inline themes/Frontend/JsBase/Theme.php]: The field is not null

File: themes/Frontend/JsBase/Theme.php [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+     */
+    protected $author = '__author__';
+
+    /**
+     * License of the theme source code.
+     *
+     * @var null
```
Comment (shyim) [inline themes/Frontend/JsBase/Theme.php]: same as above

File: themes/Frontend/JsBase/Theme.php [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+     * Defines the human readable theme name
+     * which displayed in the backend
+     * @var string
+     */
+    protected $name = '__theme_name__';
+
+    /**
+     * Allows to define a description text
+     * for the theme
+     * @var null
```
Comment (shyim) [inline themes/Frontend/JsBase/Theme.php]: same as above

File: themes/Frontend/JsBase/Theme.php [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+
+namespace Shopware\Themes\JsBase;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Shopware\Components\Form as Form;
+use Shopware\Components\Theme\ConfigSet;
+
+class Theme extends \Shopware\Components\Theme
+{
+    protected $extend = 'Bare';
```
Comment (xf-) [inline themes/Frontend/JsBase/Theme.php]: phpdoc missing

File: themes/Frontend/JsBase/Theme.php (hunk 5 above)
Comment (wolv-dev) [reply themes/Frontend/JsBase/Theme.php]: Same in themes/Frontend/Responsive/Theme.php

File: themes/Frontend/JsBase/Theme.php (hunk 6 above)
Comment (wolv-dev) [reply themes/Frontend/JsBase/Theme.php]: Same in themes/Frontend/Responsive/Theme.php

File: themes/Frontend/JsBase/Theme.php (hunk 8 above)
Comment (wolv-dev) [reply themes/Frontend/JsBase/Theme.php]: Same in themes/Frontend/Responsive/Theme.php

File: themes/Frontend/JsBase/Theme.php (hunk 9 above)
Comment (wolv-dev) [reply themes/Frontend/JsBase/Theme.php]: Same in themes/Frontend/Responsive/Theme.php

File: themes/Frontend/JsBase/Theme.php (hunk 7 above)
Comment (wolv-dev) [reply themes/Frontend/JsBase/Theme.php]: Same in themes/Frontend/Responsive/Theme.php

File: themes/Frontend/JsBase/Theme.php (hunk 8 above)
Comment (shyim) [reply themes/Frontend/JsBase/Theme.php]: see https://github.com/shopware/shopware/pull/982 :)

File: themes/Frontend/JsBase/Theme.php (hunk 9 above)
Comment (shyim) [reply themes/Frontend/JsBase/Theme.php]: see https://github.com/shopware/shopware/pull/982 :)

**Comment by wolv-dev** [comment]
Moved to https://github.com/shopware/shopware/pull/1295 -> update to version 5.3 and conflicts resolved

**Pull request closed without merging**